File: proxy/tests/script_engine/resp_set_body_based_on_req.lua
pcall(require, "../../script_libs/lua/roxy.lua")
---@type Extension
local req_set_body_based_on_req = {
	response = function(flow)
		if flow.request.url.host == "example.com" then
			flow.response.body.text = "intercepted"
		end
	end,
}
Extensions = { req_set_body_based_on_req }

File: proxy/tests/script_engine/resp_set_body_based_on_req.js
/// <reference path="../../script_libs/js/index.d.ts" />
/** @type {Extension} */
const resp_set_body_based_on_req = {
  response(flow) {
    if (flow.request.url.host === "example.com") {
      flow.response.body.text = "intercepted"
    }
  }
};
globalThis.extensions = [resp_set_body_based_on_req];
